src/libraries/System.Runtime.Caching/tests/System.Runtime.Caching/HostFileChangeMonitorTest.cs

```diff
@@ -247,5 +245,46 @@ public void UniqueId()
                 CleanupMonitoring(setup);
             }
         }
+
+        [OuterLoop]
+        [Fact]
+        public void Reasonable_Delay()
+        {
+            Tuple<string, string, string, IList<string>> setup = null;
+            try
+            {
+                setup = SetupMonitoring(nameof(Reasonable_Delay));
+
+                using var monitor = new HostFileChangeMonitor(setup.Item4);
+                var policy = new CacheItemPolicy
+                {
+                    ChangeMonitors = { monitor }
+                };
+                var config = new NameValueCollection();
+                var mc = new PokerMemoryCache("MyCache", config);
+
+                mc.Set("key", "value", policy);
+
+                // Verify the cache item is set
+                Assert.Equal("value", mc["key"]);
+
+                // Update the file dependency
+                File.WriteAllText(setup.Item2, "I am the first file. Updated.");
+
+                // Wait for the monitor to detect the change - 5s should be more than enough
+                for (int i = 0; i < 250; i++)
+                {
+                    if (!mc.Contains("key"))
+                        break;
+                    Thread.Sleep(20);
+                }
```

Comment:
When there are a lot of tests running in parallel (which xunit attempts to achieve), there can be a lot of thread contention and this might not return within 20ms. I would create a Stopwatch before the loop and use a while loop checking if the required amount of time has elapsed yet. This would be immune to CPU contention causing the test to take significantly longer due to the iteration count.

Reply:
I thought about that. Decided to just keep it simple though. The exact timing isn't important. The cache item should be removed relatively quickly - just not instantly, which is why we check on it in a loop. I'd be shocked if it ever took longer than 1s, but I wanted to be flexible enough so the test isn't noisy. 5s felt like an eternity for this, but in reality, the condition for breaking the loop should be met much, much sooner... and what that exact time is isn't important.

Do you think the potential for running so much longer than 5s that it becomes a problem in the test harness is a realistic problem?